std/algorithm/iteration.d

```diff
@@ -1463,52 +1380,178 @@ private struct GroupByImpl(alias pred, EquivRelation equivRelation, Range)
     @property bool empty() { return r.empty; }
 
     @property auto front()
-    in
     {
-        import core.exception : RangeError;
-        if (r.empty) throw new RangeError();
-    }
-    body
-    {
-        return GroupByChunkImpl!(pred, equivRelation, Range)(r, _prev);
+        return GroupByChunkImpl!(pred, Range)(r, _prev);
     }
 
     void popFront()
     {
         while (!r.empty)
         {
-            static if (equivRelation)
+            if (!fun(_prev, r.front))
             {
-                if (!fun(prev, r.front))
-                {
-                    savePrev();
-                    break;
-                }
-                r.popFront();
+                _prev = r.front;
+                break;
             }
-            else
+            r.popFront();
+        }
+    }
+}
+
+// Andrei's implementation of groupBy for forward ranges.
```

Comment:
no need for this attribution - refer to it functionally, e.g. "implementation that minimizes redundant searches"